### ISSUE TITLE
[BUG] Duplicate change events for autocomplete_input

### ISSUE THREAD

**Issue by jgriffi6** [bokeh/bokeh#11670]
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
bokeh 2.4.0
python 3.5 and 3.9.7
OS: Mac OSX and CENT OSX
Firefox and Chrome

#### Description of expected behavior and the observed behavior
Expect Autocomplete callback on_change(value) to be called twice: once when the user inputs a value, and, since the callback clears the value of the autocomplete object as a convenience to the user, a second time with an empty string.  

Behaves as expected when Autocomplete.restrict=True or when Autocomplete.restrict=False and the user supplies an un-restricted string.  Unexpected behavior: when user supplies a restricted string, the call back is called 4 times.  


#### Complete, minimal, self-contained example code that reproduces the issue

```
from bokeh.plotting import curdoc
from bokeh.layouts import layout
from bokeh.models import AutocompleteInput,Div

def my_cb(attr,old,new):
    div.tags[0]+=1
    div.text='Callback count: {:,}'.format(div.tags[0])
    #this will cause an extra-callback                                                                                                                                                                                                                                                   
    #that I could easily block                                                                                                                                                                                                                                                           
    auto_complete.value=''
    pass

completions=['Apple','Pear','Banana']
div=Div(text='',width=400,name='my_div',tags=[0])
auto_complete=AutocompleteInput(title='Restricted',name='Restricted',
                                completions=completions,
                                restrict=False,
                                placeholder='Type Dog, fine, Apple duplicate callback')
auto_complete.on_change('value',my_cb)

l=layout(auto_complete,div)
curdoc().add_root(l)


```

#### Stack traceback and/or browser JavaScript console output
None
#### Screenshots or screencasts of the bug in action

After typing restricted string 'Apple':
![image](https://user-images.githubusercontent.com/22986801/134449597-ea15a963-7fc7-420a-a023-fa8a87541234.png)

After typing un-restring string 'Dog' (new session):
![image](https://user-images.githubusercontent.com/22986801/134449649-3e70a406-005f-4e2d-bb26-e54d5c356681.png)


*Note: Everyone interacting in the Bokeh project's codebases, issue trackers, and discussion forums is expected to follow the [Code of Conduct](https://github.com/bokeh/bokeh/blob/master/CODE_OF_CONDUCT.md).*

**Comment by jgriffi6** [comment]
I've edited the bokeh-widgets.min.js on my machines in a way that should resolve the issue, and not break code for anybody else.  In readable js, the fix would be:

https://github.com/bokeh/bokeh/blob/branch-3.0/bokehjs/src/lib/models/widgets/autocomplete_input.ts#L44

```
if (!this.model.restrict) { 
```
to
```
else if (!this.model.restrict) {
```

There should be no need to call ```super.change_input()``` when the input has already been updated in the first if statement.  That is when the use supplies a string in the completions list.  Admittedly, I bring this out by clearing the string in my callback.  What is normally a harmless extra call of ```super.change_input()``` is no longer harmless.

Let me know if you'd like any tests or further information.
